py/selenium/webdriver/common/selenium_manager.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+import re
+import subprocess
+import sys
+from pathlib import Path
+from typing import Tuple
+
+from selenium.common.exceptions import WebDriverException
+
+logger = logging.getLogger(__name__)
+
+
+class SeleniumManager:
+    """
+    Wrapper for getting information from the Selenium Manager binaries.
+    """
+
+    @staticmethod
+    def get_binary() -> Path:
+        """
+        Determines the path of the correct Selenium Manager binary.
+        :Returns: The Selenium Manager executable location
+        """
+        directory = sys.platform
+        if directory == "linux":
+            directory = "linux"
+        elif directory == "darwin":
+            directory = "macos"
+        elif directory in ("win32", "cygwin"):
+            directory = "windows"
+
+        file = "selenium-manager.exe" if directory == "windows" else "selenium-manager"
+
+        path = Path(__file__).parent.joinpath(directory, file)
+
+        if not path.is_file():
+            raise WebDriverException("Unable to obtain Selenium Manager")
```

Comment:
can we improve the error here around files not found, is `WebDriverException` appropriate here vs a `FileNotFoundError` or similar?

Reply:
Maybe, but that seems weird since the code is what is trying to generate the location of the file. (as in "unable to create file" vs "unable to find file"). Everything is getting rescued right now. When/if these get exposed to the user we should re-think them.